components/button/README.md

```diff
@@ -0,0 +1,31 @@
+### Rationale
+Buttons are used to direct the user to perform a specific interaction. The copy used on the button should be clear, concise and direct.
+
+## Variants
+
```

Comment:
@armstrongb Could add all the variants of button style here. Although they are listed below